infrastructure/src/main/java/org/corfudb/infrastructure/LayoutServer.java

```diff
@@ -168,60 +175,146 @@ public synchronized void handleMessageLayoutBootstrap(
         }
     }
 
+    private boolean isSameEpoch(
+            long messageEpoch,
+            @NonNull CorfuMsg msg,
+            ChannelHandlerContext ctx,
+            @NonNull IServerRouter router) {
+        long serverEpoch = getServerEpoch();
+
+        // Check if the propose is for the correct epoch
+        if (messageEpoch != serverEpoch) {
```

Comment:
This condition already checked in NettyServerRouter, you don't need it https://github.com/CorfuDB/CorfuDB/blob/68b807519caba698ca77531a13927fdd9140bdcd/infrastructure/src/main/java/org/corfudb/infrastructure/NettyServerRouter.java#L104

Reply:
We can look into removing it. I tried to not change the code too much.